sqe-http-api/Controllers/UserController.cs

```diff
@@ -20,34 +23,126 @@ public UserController(IUserService userServiceAuthenticate)
         }
+        
+        /// <summary>
+        /// Provides the user details for a user with valid JWT in the Authorize header
+        /// </summary>
+        [HttpGet]
+        public ActionResult<UserDTO> GetCurrentUser()
+        {
+            var user = _userService.GetCurrentUser();
+
+            if (user == null)
+                return Unauthorized(new { message = "No current user", code =601 }); // TODO: Add Error Code
+
+            return user;
+        }
+        
+        /// <summary>
+        /// Creates a new user with the submitted data.
+        /// </summary>
+        /// <param name="userInfo">A JSON object with all data necessary to create a new user account</param>
+        /// <returns>Returns a UserDTO for the newly created account. Throws an HTTP 409 if the
+        /// username or email is already in use.</returns>
+        [AllowAnonymous]
+        [HttpPost]
+        public async Task<ActionResult<UserDTO>> CreateNewUser([FromBody] NewUserDTO userInfo)
+        {
+            try
+            {
+                return await _userService.CreateNewUserAsync(userInfo);
+            }
+            catch(DbDetailedFailedWrite err)
+            {
+                return Conflict(new {message = err.Message}); // I would rather not tell which element is wrong (again someone might phish for valid emails), but maybe it is not a problem.
+            }
+        }
+        
+        /// <summary>
+        /// Confirms creation of new user account.
+        /// </summary>
+        /// <param name="payload">JSON object with token from user registration email.</param>
+        /// <returns>Returns 200 for success and 404 for failure.</returns>
+        [AllowAnonymous]
+        [HttpPost("confirm-registration")]
+        public async Task<ActionResult<UserDTO>> ConfirmUserRegistration([FromBody] EmailTokenDTO payload)
+        {
+            try
+            {
+                await _userService.ConfirmUserRegistrationAsync(payload.token);
+                return Ok();
+            }
+            catch
+            {
+                return NotFound();
+            }
+        }
 
         /// <summary>
         /// Provides a JWT bearer token for valid username and password
         /// </summary>
         /// <param Name="userParam">JSON object with a username and password parameter</param>
         [AllowAnonymous]
-        [HttpPost("login")] // api/v1/user/login
+        [HttpPost("login")]
         public async Task<ActionResult<LoginResponseDTO>> AuthenticateAsync([FromBody]LoginRequestDTO userParam)
         {
             var user = await _userService.AuthenticateAsync(userParam.userName, userParam.password);
             if (user == null)
-                return Unauthorized(new { message = "Username or password is incorrect", code = 600}); // TODO: Add Error Code
+                return Unauthorized(new { message = "Username or password is incorrect ", code = 600}); // TODO: Add Error Code
```

Comment:
You added an unnecessary space. Also, the error code has already been added, so you can delete the TODO

This serves as the petty review comment of the day

Reply:
Thanks.

Done.